Sources/TripKit/server/TKBuzzInfoProvider+Routes.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  TKBuzzInfoProvider+Routes.swift
+//  TripKit
+//
+//  Created by Adrian Schönig on 27/10/2022.
+//  Copyright © 2022 SkedGo Pty Ltd. All rights reserved.
+//
+
+import Foundation
+
+extension TKBuzzInfoProvider {
+  
+  /// Fetches a list of routes in for the provided region, optionally filtered
+  ///
+  /// - warning: Calling this method without any of the filter parameters can result both in a
+  ///   slow, big response due to the large number of routes in certain regions.
+  ///
+  /// - Parameters:
+  ///   - region: The region for which to fetch a list of routes
+  ///   - query: Optional search string to filter routes by their short name (complete matches only)
+  ///           or by their name (partial matches)
+  ///   - modes: If provided, only routes using any of these mode identifiers will be returned, e.g., `pt_pub_bus`
+  ///   - operatorID: If provided, only routes for this operator will be returned
+  /// - Returns: List of routes
+  public static func fetchRoutes(forRegion region: TKRegion, query: String? = nil, modes: [String] = [], operatorID: String? = nil) async throws -> [TKAPI.Route] {
+    var paras: [String: Any] = [
+      "region": region.name
+    ]
+    
+    paras["query"] = query
```

Comment:
didn't know that there was a `query` param

are `parameters` here sent as payload instead of query params?

Reply:
That depends on the HTTP method. `info/routes.json` is using `POST` which then sends those `parameters` as the HTTP body (encoded as JSON). If it was `GET` it would send `parameters` as URL query parameters. 

`"query"` is probably a bit confusing here as a parameter as it's not a URL query parameter, but just a JSON field that happens to use that same name. See API docs: https://developer.tripgo.com/specs/#tag/Public-Transport/paths/~1info~1routes.json/post